### PR TITLE
Increase Database Source SELECT Batch Size

### PR DESCRIPTION
**Pull request by evantahler**
https://github.com/airbytehq/airbyte/pull/12400 added dynamic batch sizes to our SQL sources to prevent OOM problems.  However, I believe that we are currently creating batch sizes that are too small for _optimal speed_.  This is validated in 2 ways:
* We have shown that Airbyte's [speed increases with wider rows](https://airbytehq-team.slack.com/archives/C048E3BK1KM/p1668639622311289)
* We are not seeing OOM errors on our database sources any more, and we are not utilizing more than 60% of available memory in our source containers

unnamed

This PR does 2 things:
1. Bumps the total allocation of the source's available memory for the JDBC buffer from 50% to 60%
2. Rather than assuming the worst-case UTF8 byte-size for RECORD strings (4-byte/char), we move to 3-byte/char which is likely more common - assuming most chars fit in the ASCII space.

### CODE REVIEW

**Comment by evantahler** [comment]
@edgao I'm network-bound at home testing this out... what's the process for getting a connector tested on cloud or a dev server?

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/streaming/FetchSizeConstants.java [hunk 1]
```diff
@@ -10,7 +10,7 @@ public final class FetchSizeConstants {
   // This size is not enforced. It is only used to calculate a proper
   // fetch size. The max row size the connector can handle is actually
   // limited by the heap size.
-  public static final double TARGET_BUFFER_SIZE_RATIO = 0.5;
+  public static final double TARGET_BUFFER_SIZE_RATIO = 0.6;
```
Comment (davinchia) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/streaming/FetchSizeConstants.java:13]: curious, why not go higher?

if primary memory usage is due to this buffer, I think we can go as high as 80%. Usual java overhead is ~~10% so a 20% buffer seems good enough.

Comment (evantahler) [reply airbyte-db/db-lib/src/main/java/io/airbyte/db/jdbc/streaming/FetchSizeConstants.java:13]: I think we can keep ramping this up... but in an environment where I can't robustly test this, I want to go step-by-step

**Comment by evantahler** [comment]
@davinchia / @edgao I'm struggling to figure out how to test this.  Any ideas?

**Comment by evantahler** [comment]
I've done some local testing and thing seem ok... @edgao / @davinchia please let me know what I should do to be confident that this is safe to merge in

**Comment by davinchia** [comment]
@evantahler I might regret saying this - I'm pretty confident this is safe to merge in. We might see some OOM if not. We give all jobs up to 2GB now. This is plenty of airspace, which makes me think the chance is low.

**Comment by evantahler** [comment]
@davinchia i feel more or less the same.  @edgao how do you feel about YOLOing this one (after code freeze, of course)?

**Comment by edgao** [comment]
Seems fine to me. Vast majority of source-postgres cloud syncs are way below 2G, with just a handful of outliers

image

fyi @subodh1810 as random db sources representative

**Review by edgao — APPROVED** [review]
:shipit:

**Pull request merged**